src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.cpp

```diff
@@ -66,6 +78,156 @@ std::optional<std::string> Outflow::dg_outflow(
+
+  // The outflow condition here simply uses the outermost values on
+  // cell-centered FD grid points to compute face values on the external
+  // boundary. This is equivalent to adopting the piecewise constant FD
+  // reconstruction for FD cells at the external boundaries.
+  //
+  // In the future we may want to use more accurate methods (for instance,
+  // one-sided characteristic reconstruction using WENO) for imposing
+  // higher-order outflow boundary condition.
+  //
+  auto lapse_at_boundary = evolution::dg::subcell::slice_tensor_for_subcell(
+      lapse, subcell_extents, 1, direction);
+  auto shift_at_boundary = evolution::dg::subcell::slice_tensor_for_subcell(
+      shift, subcell_extents, 1, direction);
+
+  Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>>>
+      buffer_vars{num_face_pts};
+
+  auto& normal_dot_shift = get<::Tags::TempScalar<0>>(buffer_vars);
+  dot_product(make_not_null(&normal_dot_shift),
+              outward_directed_normal_covector, shift_at_boundary);
+
+  if (face_mesh_velocity.has_value()) {
+    auto& normal_dot_mesh_velocity = get<::Tags::TempScalar<1>>(buffer_vars);
+    dot_product(make_not_null(&normal_dot_mesh_velocity),
+                outward_directed_normal_covector, face_mesh_velocity.value());
+    get(normal_dot_shift) += get(normal_dot_mesh_velocity);
+  }
+  // The characteristic speeds are bounded by \pm \alpha - \beta_n, and
+  // saturate that bound, so there is no need to check the hydro-dependent
+  // characteristic speeds.
+  min_char_speed =
+      std::min(min(-get(lapse_at_boundary) - get(normal_dot_shift)),
+               min(get(lapse_at_boundary) - get(normal_dot_shift)));
+
+  if (min_char_speed < 0.0) {
+    ERROR("Subcell outflow boundary condition violated. Speed: "
+          << min_char_speed << "\nn_i: " << outward_directed_normal_covector
+          << "\n");
+  } else {
+    // Once the outflow condition has been checked, we fill each slices of the
+    // ghost data with the boundary values. The reason that we need this step is
+    // to prevent floating point exceptions being raised while computing the
+    // subcell time derivative because of NaN or uninitialized values in ghost
+    // data.
+
+    using RestMassDensity = hydro::Tags::RestMassDensity<DataVector>;
+    using Pressure = hydro::Tags::Pressure<DataVector>;
+    using LorentzFactorTimesSpatialVelocity =
+        hydro::Tags::LorentzFactorTimesSpatialVelocity<DataVector, 3>;
+    using MagneticField = hydro::Tags::MagneticField<DataVector, 3>;
+    using DivergenceCleaningField =
+        hydro::Tags::DivergenceCleaningField<DataVector>;
+
+    using prim_tags_for_reconstruction =
+        tmpl::list<RestMassDensity, Pressure, LorentzFactorTimesSpatialVelocity,
+                   MagneticField, DivergenceCleaningField>;
+
+    // Create a single large DV to reduce the number of Variables allocations
+    const size_t buffer_size_per_grid_pts =
```

Comment:
Not sure this is the best way to count the number of individual components of `<prim_tags_for_reconstruction>`..